packages/@aws-cdk/aws-s3/lib/rule.ts

```diff
@@ -142,6 +142,38 @@ export interface LifecycleRule {
   readonly objectSizeGreaterThan?: number;
 }
 
+/**
+ * A point in time when an object should transition to a different storage class.
+ */
+export class TransitionMoment {
+
+  /**
+   * A moment defined by a specific date.
+   *
+   * The date value must be in ISO 8601 format. The time is always midnight UTC.
+   */
+  public static at(date: Date): TransitionMoment {
+    return { transitionDate: date };
+  }
+
+  /**
+   * A moment defined by the number of days after the object creation.
+   */
+  public static after(duration: Duration): TransitionMoment {
```

Comment:
Given that this is specifically supposed to be duration in days, I'm wondering if it would be better to take in a number that represents the number of days. 

Is there a use case where a user would want to input this in millis, seconds, minutes, or hours instead? It seems like allowing those units mostly opens this up for a lot more potential errors on the user's side.

On the other hand, I also see that `transitionAfter` was in the units of `Duration` so I can also see an argument for keeping that consistent.

```suggestion
  public static after(days: number): TransitionMoment {
```

Thoughts? Just adding this as a comment and not doing a review requesting changes because I'm not sure which is actually the better user experience here.

Reply:
what about leaving the type as `Duration` and doing runtime error checking that validates that it is a `Duration` in days?

Reply:
@TheRealAmazonKendra I don't want to make too many assumptions here. Maybe there is a use case in which the user consumes raw data in hours and passes it to the construct. They should benefit from the abstraction we've built for that. And, as you said, the current API already uses `Duration`. When it comes down to generating the L1 properties, it will convert to an integer number of days. If that conversion is not possible, it will fail at synth time.